### ISSUE TITLE
false positive finding for WRONG_WHITESPACE on gradle plugins block

### ISSUE THREAD

**Issue by WebTiger89** [saveourtool/diktat#1313]
The warning states:

> [WRONG_WHITESPACE] incorrect usage of whitespaces for code separation: version should have 1 space(s) before and 1 space(s) after, but has 0 space(s) after

for code in settings.gradle.kts:
```
plugins {
    id("com.gradle.enterprise") version("3.10.1") // this line
}
```
Maybe the rule can be adjusted/improved here.

**Comment by akuleshov7** [comment]
That looks like a bug actually, @WebTiger89 you can exclude `.kts` scripts while we are investigating it:

```
 configure<DiktatExtension> {
        diktatConfigFile = rootProject.file("diktat-analysis.yml")
        githubActions = findProperty("diktat.githubActions")?.toString()?.toBoolean() ?: false
        inputs {
            // using `Project#path` here, because it must be unique in gradle's project hierarchy
           exclude("**/*.kt")
        }
    }
```

**Comment by akuleshov7** [comment]
Also this issue affects comments 

```
[WRONG_WHITESPACE] incorrect usage of whitespaces for code separation: version should have 1 space(s) before and 1 space(s) after, but has 0 space(s) after
[COMMENT_WHITE_SPACE] there should be a white space between code and comment also between code start token and comment text: There should be 2 space(s) before comment text, but there are too many in // this line
```

**Comment by WebTiger89** [comment]
Thank you very much. I have found a lot of false positives. I will open new issues for them now. I hope this won't be too annoying ^^

**Comment by akuleshov7** [comment]
> Thank you very much. I have found a lot of false positives. I will open new issues for them now. I hope this won't be too annoying ^^

that would be extremely useful! You cannot even imagine, how we appreciate that

**Comment by WebTiger89** [comment]
Here is another example:
```
    val onClick: () -> Unit = remember {
        {
            when {
                ContextCompat.checkSelfPermission(
                    context,
                    permission
                ) == PackageManager.PERMISSION_GRANTED -> {
                    // You can use the API that requires the permission.
                    Log.d("PermissionComponent", "permission [$permission].")
                    onPermissionGranted()
                }
                ActivityCompat.shouldShowRequestPermissionRationale(
                    context as Activity,
                    permission
                ) -> {
                    /*
                     * In an educational UI, explain to the user why your app requires this
                     * permission for a specific feature to behave as expected. In this UI,
                     * include a "cancel" or "no thanks" button that allows the user to
                     * continue using your app without granting the permission.
                     */
                    Log.d("PermissionComponent", "show rationale.")
                    permissionState.showRationale = true
                }
                else -> {
                    // Asking for permission
                    Log.d("PermissionComponent", "ask for permission [$permission].")
                    requestPermissionLauncher.launch(permission)
                }
            }
        }
    }
```
It is the second line.

> [WRONG_WHITESPACE] incorrect usage of whitespaces for code separation: there should be a whitespace before '{'

**Comment by WebTiger89** [comment]
Thanks for the details. What about the second example?

**Comment by petertrr** [comment]
Yes, the second example is a bug in diktat. I think we usually expect opening braces to be on the same line as the previous code and therefore don't properly handle cases when a line starting with white spaces and an opening braces is indeed valid.
